sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleHistogramAggregator.java

```diff
@@ -0,0 +1,188 @@
+
+  DoubleHistogramAggregator(
+      Resource resource,
+      InstrumentationLibraryInfo instrumentationLibraryInfo,
+      InstrumentDescriptor instrumentDescriptor,
+      ImmutableDoubleArray boundaries,
+      boolean stateful) {
+    super(resource, instrumentationLibraryInfo, instrumentDescriptor, stateful);
+    this.boundaries = boundaries;
+  }
+
+  @Override
+  public AggregatorHandle<HistogramAccumulation> createHandle() {
+    return new Handle(this.boundaries);
+  }
+
+  @Override
+  public final HistogramAccumulation merge(HistogramAccumulation x, HistogramAccumulation y) {
+    if (!x.getBoundaries().equals(y.getBoundaries())) {
+      // If this happens, it's a pretty severe bug in the SDK.
+      if (!loggedMergingInvalidBoundaries) {
+        logger.log(
+            Level.SEVERE,
+            "can't merge histograms with different boundaries, something's very wrong: "
+                + "x.boundaries="
+                + x.getBoundaries()
+                + " y.boundaries="
+                + y.getBoundaries());
+        loggedMergingInvalidBoundaries = true;
+      }
+      return HistogramAccumulation.create(
+          0, 0, ImmutableDoubleArray.of(), ImmutableLongArray.of(0));
+    }
+
+    long[] mergedCounts = new long[x.getCounts().length()];
+    for (int i = 0; i < x.getCounts().length(); ++i) {
+      mergedCounts[i] = x.getCounts().get(i) + y.getCounts().get(i);
+    }
+    return HistogramAccumulation.create(
+        x.getCount() + y.getCount(),
+        x.getSum() + y.getSum(),
+        x.getBoundaries(),
+        ImmutableLongArray.copyOf(mergedCounts));
+  }
+
+  @Override
+  public final MetricData toMetricData(
+      Map<Labels, HistogramAccumulation> accumulationByLabels,
+      long startEpochNanos,
+      long lastCollectionEpoch,
+      long epochNanos) {
+    return MetricData.createDoubleHistogram(
+        getResource(),
+        getInstrumentationLibraryInfo(),
+        getInstrumentDescriptor().getName(),
+        getInstrumentDescriptor().getDescription(),
+        getInstrumentDescriptor().getUnit(),
+        DoubleHistogramData.create(
+            isStateful() ? AggregationTemporality.CUMULATIVE : AggregationTemporality.DELTA,
+            MetricDataUtils.toDoubleHistogramPointList(
+                accumulationByLabels,
+                isStateful() ? startEpochNanos : lastCollectionEpoch,
+                epochNanos)));
+  }
+
+  @Override
+  public HistogramAccumulation accumulateDouble(double value) {
+    return HistogramAccumulation.create(
+        1, value, ImmutableDoubleArray.of(), ImmutableLongArray.of(1));
+  }
+
+  @Override
+  public HistogramAccumulation accumulateLong(long value) {
+    return HistogramAccumulation.create(
+        1, value, ImmutableDoubleArray.of(), ImmutableLongArray.of(1));
+  }
+
+  static final class Handle extends AggregatorHandle<HistogramAccumulation> {
+    private final ReentrantReadWriteLock lock = new ReentrantReadWriteLock();
```

Comment:
to follow-up on the lock...if we only need write, could we just use a normal `ReentrantLock` rather than the more complex read/write version?